sql/driver/driver_test.go

```diff
@@ -45,18 +55,79 @@ func cleanup(s *server.TestServer, db *sql.DB) {
 
 func TestPlaceholders(t *testing.T) {
 	defer leaktest.AfterTest(t)
+
+	// Stolen from https://en.wikipedia.org/wiki/List_of_tz_database_time_zones
+	locationNames := []string{
+		"Canada/Newfoundland", // half-hour zone
+		"Europe/London",       // same as UTC, except for DST
+		"Pacific/Kiritimati",  // maximum positive offset
+		"Pacific/Midway",      // maximum negative offset
+		"US/Pacific",          // negative offset
+		"UTC",
+	}
+
 	s, db := setup(t)
 	defer cleanup(s, db)
 
-	year, month, day := 3015, time.August, 30
-	timeVal := time.Date(year, month, day, 3, 34, 45, 345670000, time.UTC)
-	dateVal := time.Date(year, month, day, 0, 0, 0, 0, time.UTC)
+	// We don't want to spend forever on this test.
```

Comment:
This comment, while true, is no longer very useful.

The combinatorics in this test don't seem to be worth the runtime (62 seconds in race mode on CircleCI!). All it's really testing is that the truncation performed by `::date` is consistent with a local call to `driver.MakeDate`, and only a few of the combinations actually have the potential to show a bug there. I think we should continue to run TestPlaceholders with a single timestamp, in a single non-UTC time zone (where the timeVal is chosen so that truncating to a date in UTC would give the wrong result).

Separately, we can do cheaper tests of some of the interesting combinations (we just need one pair of summer/winter times, not one for every combination of year and hour), including writing a value in one time zone and reading it back in another. If we had a timestamp-to-string function that used the session time zone (do we?) these tests would be easier to write as logic tests.